processor/src/main/java/org/mapstruct/ap/internal/util/MapperConfiguration.java

```diff
@@ -70,6 +97,36 @@ private MapperConfiguration(MapperPrism mapperPrism) {
             this.config = null;
             this.mapperConfigPrism = null;
         }
+
+        AnnotationMirror componentModelAnnotation = null;
+        for ( AnnotationMirror am : e.getAnnotationMirrors() ) {
+            ComponentModelMapperPrism cm = ComponentModelMapperPrism.getInstanceOn( am.getAnnotationType()
+                .asElement() );
+            if ( cm != null ) {
+                if ( componentModelAnnotation == null ) {
+                    componentModelAnnotation = am;
+                }
+                else {
+                    messager.printMessage( e, am, Message.COMPONENT_MODEL_MULTIPLE_MODELS );
+                }
+            }
+        }
+        this.componentModelAnnotation = componentModelAnnotation;
+
+        // warning if componentModel attribute and annotation have different model values
+        if ( mapperPrism.values.componentModel() != null && componentModelAnnotation != null ) {
+            String annotationValue = ComponentModelMapperPrism.getInstanceOn(
+                componentModelAnnotation.getAnnotationType().asElement()
+            ).value();
+            if ( !mapperPrism.componentModel().equalsIgnoreCase( annotationValue ) ) {
+                messager.printMessage(
+                    e,
+                    componentModelAnnotation,
+                    Message.COMPONENT_MODEL_PROPERTY_NOT_MATCHING_ANNOTATION,
+                    mapperPrism.componentModel()
+                );
+            }
+        }
```

Comment:
I added the validation of the mapper here, there is also a TODO a few lines earlier that some more validations are required here. Removing the `printMessage` calls here would allow to remove the `configurations` cache.